### PR TITLE
Migrate to app routing

### PR DESCRIPTION
**Pull request by Ben0189**
Migrate barebone starter from Page router -> App router

### CODE REVIEW

File: app/[...filename]/client-page.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+interface ClientPageProps {
+    query: string;
+    variables: {
+      relativePath: string;
+    };
+    data: {page: PageQuery["page"]};
+}
+
+
```
Comment (ncn-ssw) [inline app/[...filename]/client-page.tsx:14]: **nitpick**: Remove extra newline

File: app/[...filename]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export async function generateStaticParams() {
+  const pages = await client.queries.pageConnection();
+  const paths = pages.data?.pageConnection?.edges?.map((edge) => ({
+    filename: edge?.node?._sys.breadcrumbs,
+  }));
+
+  return paths || [];
+}
+
+
```
Comment (ncn-ssw) [inline app/[...filename]/page.tsx:13]: **nitpick:** Remove extra newline

File: app/posts/[...filename]/client-page.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+"use client"
+import { useTina } from "tinacms/dist/react";
+import { PostQuery } from "../../../tina/__generated__/types";
+interface ClientPageProps {
```
Comment (ncn-ssw) [inline app/posts/[...filename]/client-page.tsx:4]: **issue**: There should be vertical space after imports and after interface declarations - please investigate why ESLint is not detecting style issues.

File: app/[...filename]/client-page.tsx (hunk 1 above)
Comment (Ben0189) [reply app/[...filename]/client-page.tsx]: Done

File: app/[...filename]/page.tsx (hunk 2 above)
Comment (Ben0189) [reply app/[...filename]/page.tsx]: Done

File: app/posts/[...filename]/client-page.tsx (hunk 3 above)
Comment (Ben0189) [reply app/posts/[...filename]/client-page.tsx:5]: Added eslint rules

**Review by ncn-ssw — APPROVED** [review]
🦙

**Pull request merged**